views/js/pciCreator/dev/audioRecordingInteraction/runtime/audioRecordingInteraction.js

```diff
@@ -228,6 +244,54 @@ define([
             });
         },
 
+        initRecording: function initRecording() {
+            var delayInSeconds = this.config.delayMinutes * 60 + this.config.delaySeconds;
+            var ctrCache = {};
+            var ctr;
+            var self = this;
+
+            // no auto start, don't start recording
+            if (this.config.autoStart !== true) {
+                return;
+            }
+
+            // no delay and no media stimulus, start recording now
+            if (delayInSeconds === 0 && !this.hasMediaStimulus()) {
+                this.startRecording();
+                return;
+            }
+
+            // cache controls states
+            for (ctr in this.controls) {
+                if (this.controls.hasOwnProperty(ctr)) {
+                    ctrCache[ctr] = this.controls[ctr].getState();
+                    this.controls[ctr].disable();
+                }
+            }
+
+            // cleaning up delay callback
+            this._cleanDelayCallback();
+
+            // adding a delay before start recording...
+            this._delayCallback = setTimeout(function() {
+
+                // restore controls states
+                for (ctr in ctrCache) {
```

Comment:
Prefer lodash `forEach` instead of `for..in` loops.

Reply:
ok.